docusaurus/docs/tutorial/03.md

```diff
@@ -1,214 +1,449 @@
-  const newState = { ...state };
-
-  switch (action.type) {
-    case "ADD_TODO":
-        return {
-            ...state,
-            term: "",
-            history: state.item.push(state.term)
-        };
-    default:
-      return state;
-  }
-};
+export default TodoItem;
+```
+Don't worry if it's long. If you read it carefully, we add every mutation into a single `<Mutation>` component. That's why there are 3 nested Mutation out there. Also we add some network error handler like `{loading || (deleteLoading || uploadLoading) ...}` and `{error || (deleteError || updateError) ...}`. That makes our items contain 3 mutations `updateTodo`, `updateStatus`, and `deleteTodo`.
+
+3. To make these mutation works, of course we also need to make the list stored in our GraphQL server. If it's stored in the server, then we need to fetch it when rendering our `todo-list` component. We're going to use `<Query>` components to get our data from GraphQL server in `todo-list.js`.
+```javascript
+/* src/component/todo-list/todo-list.js */
+
+import React from "react";
+
+import { Query } from "@treats/graphql";
+
+import { todoQuery } from "../../_graphql/todo";
+
+import TodoItem from "../todo-item";
+
+const TodoList = () => (
+    <Query 
+        query={todoQuery.GetAllTodoes}
+        ssr={false}
+        fetchPolicy="cache-and-network"
+    >
+        {
+            ({ loading, error, data }) => {
+                if (loading) return <p>Loading....</p>
+                if (error) return <p style={{"color": "red"}}>ERROR</p>
+                if (data.allTodoes) {
+                    return (
+                        <div>
+                            {data.allTodoes.map(item => (
+                                <TodoItem key={item.id} {...item} />
+                            ))}
+                        </div>
+                    );
+                }
+                return (
+                    <div>
+                        No Data
+                    </div>
+                )
+            }
+        }
+    </Query>
+);
   
-export default TodoReducer;
-
+export default TodoList;
 ```
 
-```
-/* ./src/_redux/reducer.js */
-import { combineReducers } from "@treats/redux";
+4. Yay! Now we can get data from GraphQL server. _Wait a minute... But where the server get the data from?_. We must create add `CreateTodo` mutation into our form in `todo.js`.
+```javascript
+/* src/page/todo/todo.js */
 
-import ReduxpageReducer from "./reduxpage";
-import TodoReducer from "./todo";
+import React from "react";
+import { FormattedMessage, injectIntl } from "@treats/intl";
+import Link from "@treats/component/link";
+import { connect } from "@treats/redux";
+import { Mutation } from "@treats/graphql";
+
+import { todoMutation, todoQuery } from "../../_graphql/todo";
+
+import TodoList from "../../component/todo-list";
+
+import style from "./todo.css";
+
+const Todo = ({ intl, name }) => (
+    <div>
+        <Link href="/profile" isPush>Change Name</Link>
```

Comment:
I think you can hide the rest of the code to give clearance and emphasize more about the current GraphQL section, so there will be <Mutation /> and <TodoList /> only.